experimental/uefi/app/src/remote_attestation.rs

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Server-side implementation of the bidirectional gRPC remote attestation handshake
+//! protocol.
+//!
+//! A simplified version of the implementation from the `grpc_unary_attestation`
+//! crate. TODO(#2741): Refactor this to share more code between the two runtimes.
+
+use alloc::vec::Vec;
+use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker};
+
+/// Number of sessions that will be kept in memory.
+const SESSIONS_CACHE_SIZE: usize = 10000;
+
+pub struct AttestationHandler<F> {
+    session_tracker: SessionTracker,
+    request_handler: F,
+}
+
+const MOCK_TEE_CERTIFICATE: [u8; 0] = [];
+const MOCK_ADDITIONAL_INFO: [u8; 0] = [];
+
+impl<F> AttestationHandler<F>
+where
+    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> Vec<u8>,
+{
+    pub fn create(request_handler: F) -> Self {
+        let session_tracker = SessionTracker::create(
+            SESSIONS_CACHE_SIZE,
+            MOCK_TEE_CERTIFICATE.to_vec(),
+            MOCK_ADDITIONAL_INFO.to_vec(),
+        );
+
+        Self {
+            session_tracker,
+            request_handler,
+        }
+    }
+
+    pub fn message(&mut self, session_id: SessionId, request: Vec<u8>) -> Vec<u8> {
+        let mut session_state = {
+            self.session_tracker
+                .pop_session_state(session_id)
```

Comment:
nit: looking at this code the naming feels slightly inaccurate. My first assumptions was that the pop would fail if the session does not exist, so I assume the code would not be able to start new sessions. Perhaps sommething lik "pop_or_create_session_state" would make it clearer that this would only fail if the session did not exist and it could not be created either.

Reply:
Good point, will update that naming

Reply:
Opened https://github.com/project-oak/oak/pull/2751 to address